src/com/google/javascript/jscomp/PolymerPass.java

```diff
@@ -89,15 +84,22 @@ public void hotSwapScript(Node scriptRoot, Node originalRoot) {
 
   @Override
   public void visit(NodeTraversal traversal, Node node, Node parent) {
-    Preconditions.checkState(
-        polymerElementExterns != null && polymerElementProps != null && globalNames != null,
-        "Cannot call visit() before process()");
+    Preconditions.checkNotNull(globalNames != null, "Cannot call visit() before process()");
+
     if (isPolymerCall(node)) {
-      rewriteClassDefinition(node, parent, traversal);
+      if (polymerElementExterns != null) {
+        rewritePolymer1ClassDefinition(node, parent, traversal);
+      } else if (!warnedPolymer1ExternsMissing) {
+        compiler.report(JSError.make(polymerElementExterns, POLYMER_MISSING_EXTERNS));
+        warnedPolymer1ExternsMissing = true;
+      }
+    } else if (isPolymerClass(node)) {
+      rewritePolymer2ClassDefinition(node, parent, traversal);
```

Comment:
Currently for the 2.0 (class based elements), the compiler doesn't need externs at this point. The 1.0 version of the pass heavily depended on them, but I'm trying to plan for the case where Polymer 2.0 can be used as either source or externs.

If the Polymer externs or source definitions are omitted, the standard compiler passes that come later will highlight the problem.